### ISSUE TITLE
Support for int64 with webgl backend of the web runtime

### ISSUE THREAD

**Issue by josephrocca** [microsoft/onnxruntime#9724]
**Is your feature request related to a problem? Please describe.**
Here's a working demo of [AnimeGANv2](https://huggingface.co/spaces/akhaliq/AnimeGANv2) with ONNX Runtime Web: https://josephrocca.github.io/anime-gan-v2-web/onnx.html The wasm backend works, but if you switch it to the webgl backend with the drop-down selector on that page, it gives the error "int64 is not supported":

![image](https://user-images.githubusercontent.com/1167575/141192627-2c0c1f2d-8365-42dd-9b93-f682481c3554.png)

I'm wondering whether it's possible for the webgl backend to support this model? [Here's the onnx model](https://github.com/josephrocca/anime-gan-v2-web/blob/main/anime-gan-v2.onnx) that I'm trying to execute.

**System information**
- ONNX Runtime version (you are using): onnxruntime-web@1.9.0

**Describe the solution you'd like**
I'd like to be able to use the webgl backend for this model (purely for performance reasons). I don't know enough about the ONNX web runtime or the webgl backend to give any details on the best way to approach a solution here.

**Describe alternatives you've considered**
The wasm backend is fine, but I'm guessing there would be a significant speedup for a model like this if it could use the GPU.

---

Thanks for your work on ONNX Runtime Web! I was pleasantly surprised to see that the wasm backend worked on my first try. Meanwhile converting the original Pytorch model to tfjs and tflite has led to some errors that I've yet to resolve.

**Comment by josephrocca** [comment]
One workaround might be to use this script to typecast to int32, since int64 seems to often be unnecessary: https://github.com/aadhithya/onnx-typecast I'm in the process of testing this but am held up by a [different bug](https://github.com/microsoft/onnxruntime/issues/9757).

**Comment by josephrocca** [comment]
I updated [the example](https://josephrocca.github.io/anime-gan-v2-web/onnx.html) to v1.11.0 and the webgl runtime still doesn't support `int64`, so I think this should remain open.

I wonder if it'd be easier to fulfill the needs of most users by just having an option/setting to automatically cast `int64` to `int32`? I think in most cases when trying to get a pre-trained model working on the web, the model uses `int64` not because it actually requires 64 bits, but because the model author just uses that as the default int.

**Comment by kethan1** [comment]
Also, the tool mentioned above ([onnx-typecast](https://github.com/aadhithya/onnx-typecast)) ends up producing an invalid output model with certain operators due to input type specifications (https://github.com/onnx/onnx/blob/main/docs/Operators.md#inputs-98). 

Automatically casting the model might run into these limitations (I don't know too much about the onnx format spec, just throwing this out there).

**Comment by leepenkman** [comment]
bump :)

**Comment by leepenkman** [comment]
Would be nice to have an estimate on when this might happen (int64 support for the reshape operator).

Theres also an expand operator + fp16 support + adaptive_avg_pool_2d we would like to see supported in onnxruntime-web+webgl runtime, so interested if theres any timeline on when those might happen too even if its a rough guesstimate

**Comment by miguel-lorenzo** [comment]
Any updates?
